core/processor.go

```diff
@@ -503,3 +515,53 @@ func (b *Processor) toLowLevelMessage(
 	}
 	return &withdrawalTx, nil
 }
+
+func (b *Processor) CheckByFilterOptions(botDelegatedWithdrawToEvent *L2ContractEvent, receipt *types.Receipt) error {
+	L2StandardBridgeBotAbi, _ := bindings2.L2StandardBridgeBotMetaData.GetAbi()
+	withdrawToEvent := bindings2.L2StandardBridgeBotWithdrawTo{}
+	indexedArgs := func(arguments abi.Arguments) abi.Arguments {
+		indexedArgs := abi.Arguments{}
+		for _, arg := range arguments {
+			if arg.Indexed {
+				indexedArgs = append(indexedArgs, arg)
+			}
+		}
+		return indexedArgs
+	}
+	err := abi.ParseTopics(&withdrawToEvent, indexedArgs(L2StandardBridgeBotAbi.Events["WithdrawTo"].Inputs), receipt.Logs[botDelegatedWithdrawToEvent.LogIndex].Topics[1:])
+	if err != nil {
+		return fmt.Errorf("parse indexed event arguments from log.topics of L2StandardBridgeBotWithdrawTo event, err: %v", err)
+	}
+
+	err = L2StandardBridgeBotAbi.UnpackIntoInterface(&withdrawToEvent, "WithdrawTo", receipt.Logs[botDelegatedWithdrawToEvent.LogIndex].Data)
+	if err != nil {
+		return fmt.Errorf("parse non-indexed event arguments from log.data of L2StandardBridgeBotWithdrawTo event, err: %v", err)
+	}
+
+	if len(b.cfg.L2StandardBridgeBot.WhitelistL2TokenList) > 0 {
+		isWhitelisted := false
+		for _, tokenAddress := range b.cfg.L2StandardBridgeBot.WhitelistL2TokenList {
```

Comment:
1. This implementation is inefficient. Plz consider maintain an private attribute `whiteListL2TokenMap` and init it when loading the config and a public `IsInWhitelist` function to check.
2. Some of the addresses may have checksum, which leads the string compare failed. Consider turning all of them to lower case before comparing.

Reply:
Regard as `#1`, it's updated at 955071e.

Regard as `#2`, 955071e transforms the token list into a map using the `l2Token: common.Address` as key, and a new test case enforces address comparison.